src/main/java/de/tum/in/www1/artemis/service/compass/CompassCalculationEngine.java

```diff
@@ -222,23 +228,19 @@ public LocalDateTime getLastUsedAt() {
     }
 
     @Override
-    public Map<Long, Grade> getModelsWaitingForAssessment() {
-        Map<Long, Grade> optimalModels = new HashMap<>();
-        for (long modelId : modelSelector.getModelsWaitingForAssessment()) {
-            optimalModels.put(modelId, getGradeForModel(modelId));
-        }
-        return optimalModels;
+    public List<Long> getModelsWaitingForAssessment() {
+        return modelSelector.getModelsWaitingForAssessment();
     }
 
     @Override
     public void removeModelWaitingForAssessment(long modelSubmissionId, boolean isAssessed) {
         modelSelector.removeModelWaitingForAssessment(modelSubmissionId);
-        if (!isAssessed && (modelIndex.getModelMap().get(modelSubmissionId) == null || !modelIndex.getModelMap().get(modelSubmissionId).isEntirelyAssessed())) {
-            modelSelector.removeAlreadyAssessedModel(modelSubmissionId);
-        }
-        else if (isAssessed) {
+        if (isAssessed) {
```

Comment:
JavaDoc missing

Reply:
The JavaDoc for this method can be found in the `CalculationEngine` interface.